### DISCUSSION TITLE
Sharing a small framework made w/ blockdom

### DISCUSSION THREAD

**Discussion by noahdamiani** [ged-odoo/blockdom#41] [Show and tell]
[github](https://github.com/rodeostar/lasso)

[demo app](https://lasso-h3smx.ondigitalocean.app/)

Hey there, just thought I would share a small framework I put together based on the tomato tutorial. It's using blockdom for client and server side rendering. The demo is pretty ugly aesthetically and is just a grid of some things a user might want to do with a frontend framework. This was just a small project for fun, it's not published to NPM yet, though its setup to be run via npx as a binary.

Things beyond the tomato tutorial:
- Async effects
- API routes (i.e. api/todos.ts will create /api/todos)
- Page Routing for the server and client (i.e. pages/index.ts -> /, pages/about.ts -> /about)
- Client side router for SPA feel
- Tailwind for quick access to styling classes via their JIT compiler
- Conditional styles
- Small shared store utility for connecting state between components

@ged-odoo Really enjoyed reading through the blockdom source and getting to know the tools in there. Thanks!

**Comment by ged-odoo** [comment]
Very cool! I intended to do a followup for the tomato framework some day, but life and work came in the way.

I just had a very quick look at your `lasso` framework, and it looks very nice. I hope you had fun doing it :)  Also, if you have any comment on the tomato framework, don't hesitate to open issues/PR
